packages/gatsby/src/redux/actions.js

```diff
@@ -87,6 +98,62 @@ actions.createPage = (page, plugin = ``, traceId) => {
 }
 
 /**
+ * Delete a layout
+ * @param {string} layout a layout object with at least the name set
+ * @example
+ * deleteLayout(layout)
+ */
+actions.deleteLayout = (layout, plugin = ``) => {
+  return {
+    type: `DELETE_LAYOUT`,
+    payload: layout,
+  }
+}
+
+/**
+ * Create a layout.
+ * @param {Object} layout a layout object
+ * @param {string} layout.id Unique id for layout
+ * @param {string} layout.component The absolute path to the component for this layout
+ * @example
+ * createLayout({
+ *   id: `myNewLayout`,
```

Comment:
this is asymetrical with the Page api yeah? I'm not sure what to do with that, but something to think about...is `id` valuable? the path to the layout is gonna be unique enough right?